scipy/_lib/_array_api.py

```diff
@@ -12,6 +12,7 @@
 import warnings
 
 import numpy as np
+from numpy.testing import assert_
```

Comment:
We have a preference for not using this `assert_` and instead just using the plain `pytest` `assert` at this point, though I forget where we documented that.

In any case, please don't make that change at this point, I'm just doing a hopefully-final pass...

Reply:
https://github.com/scipy/scipy/pull/19186#discussion_r1320077302 makes sense, but we have a line break, so I think we want `assert_`? Or `msg=...` on the line beforehand might be better.

Reply:
I'd just split the message assignment yeah, but not worth it at this point of code review IMO.

Reply:
Yup, I noted this earlier, but let it pass to avoid the explicit line breaks.